city_scrapers/spiders/il_pollution_control.py

```diff
@@ -0,0 +1,164 @@
+import json
+from datetime import datetime
+from urllib.parse import urljoin
+
+import scrapy
+from city_scrapers_core.constants import BOARD, CANCELLED, FORUM, NOT_CLASSIFIED, PASSED, TENTATIVE
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class IlPollutionControlSpider(CityScrapersSpider):
+    name = "il_pollution_control"
+    agency = "Illinois Pollution Control Board"
+    timezone = "America/Chicago"
+    allowed_domains = ["pcb.illinois.gov"]
+    start_urls = ["https://pcb.illinois.gov/ClerksOffice/MeetingMinutes"]
+    json_url = "https://pcb.illinois.gov/ClerksOffice/GetCalendarEvents"
+
+    def __init__(self, *args, **kwargs):
+        self.link_map = dict()  # Populated by self._parse_minutes()
+        self.link_parse_complete = False
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def from_crawler(cls, crawler, *args, **kwargs):
+        """ Overridden `from_crawler` to connect `spider_idle` signal. """
+        spider = super(IlPollutionControlSpider, cls).from_crawler(crawler, *args, **kwargs)
+        crawler.signals.connect(spider.spider_idle, signal=scrapy.signals.spider_idle)
+        return spider
+
+    def spider_idle(self):
+        """ React to `spider_idle` signal by starting JSON parsing after _parse_minutes."""
+        if not self.link_parse_complete:
+            self.crawler.signals.disconnect(self.spider_idle, signal=scrapy.signals.spider_idle)
+            self.crawler.engine.crawl(
+                scrapy.Request(self.json_url, callback=self._parse_json), self
+            )
+            self.link_parse_complete = True
+            raise scrapy.exceptions.DontCloseSpider
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+        """
+        for item in response.xpath("//iframe/@src"):
+            yield scrapy.Request(item.get(), callback=self._parse_minutes)
+
+    def _parse_minutes(self, response):
+        """ Traverse tree of URLs and populate self.link_map """
+        for item in response.xpath("//td[@class='name']/a"):
```

Comment:
I don't think we want to scrape all the way back through the archive every time, since we'd only need to pull that information once. It would be great to just pull minutes for the past two years displayed

Reply:
Sure thing, will do.